src/FlyCube/Instance/BaseTypes.h

```diff
@@ -408,6 +408,7 @@ struct BindKey {
     uint32_t space = 0;
     uint32_t count = 1;
     uint32_t remapped_slot = ~0;
+    bool is_root_constant = false;
```

Comment:
I suggest to use `ViewType::kPushConstant` instead.

Reply:
Actually no, it's better to use `ViewType::kConstantBuffer`. Because `ShaderReflection` must be able to distinguish `ViewType::kPushConstant` from `ViewType::kConstantBuffer`, but there is no way to do so in dxil, since dxc ignore `[[vk::push_constant]]` when compile to dxil.

But since we still need to pass the size for push_constant, it is better to pass it separately from the BindKey.

Reply:
If I use `ViewType::kConstantBuffer` instead of this boolean, how can I know if the root BindKey should go to a root constant?

Reply:
For example, we can separate them.

```
struct PushConstant {
    ShaderType shader_type = ShaderType::kUnknown;
    uint32_t size = 0;
};

std::shared_ptr<BindingSetLayout> CreateBindingSetLayout(const std::vector<BindKey>& descs, const std::vector<PushConstant>& push_constants = {});
```

Reply:
Not sure what is best to use, just ShaderType or BindKey.

1) ShaderType
At most 1 push constant per shader stage to match Vulkan limitation.
Slot and space are 0 because spirv do not save bindings when [[vk::push_constant]] is used.
2) BindKey
Allow multiple push constant in DirectX12 and Metal, but Vulkan still supports at most 1 push constant per shader stage at slot/space 0 with using [[vk::push_constant]].
More flexible, but harder to use in Vulkan.